src/rank_llm/rerank/rankllm.py

```diff
@@ -101,7 +101,7 @@ def sliding_windows(
         start_pos = rank_end - window_size
         prompts = []
         permutations = []
-        while start_pos >= rank_start:
+        while end_pos > rank_start and start_pos + step != rank_start:
```

Comment:
I had to think about why the second condition was needed. please add a comment explaining the rationale

Reply:
Done!